BoltApp/bolt-magento2
# M2P-450 Remove legacy unit test flow from Circle CI

# Description

- Remove jobs like unit-php74-magento24 and add jobs like integration-magento-php74-magento24 instead

- Remove function skipTestInUnitTestsFlow

- Update ReadMe https://github.com/BoltApp/bolt-magento2/blob/master/Test/Unit/README.md

- Check if we calculate test coverage correctly

Fixes: https://boltpay.atlassian.net/browse/M2P-450

#changelog Remove legacy unit test flow from Circle CI

# Type of change



# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.
